lib/adb.js

```diff
@@ -1,1700 +1,66 @@
-      if (err || stderr) {
-        logger.warn(stderr);
-        return cb(new Error("hasInternetPermissionFromManifest failed. " + err));
-      }
-      var hasInternetPermission = new RegExp("uses-permission:.*'android.permission.INTERNET'").test(stdout);
-      cb(null, hasInternetPermission);
-    });
-  }.bind(this));
-};
-
-ADB.prototype.reboot = function (cb) {
-  var adbCmd = "stop; sleep 2; setprop sys.boot_completed 0; start";
-  this.shell(adbCmd, function (err) {
-    if (err) return cb(err);
-    var bootCompleted = false;
-    var i = 90;
-    logger.debug('waiting for reboot, this takes time.');
-    async.until(
-      function test() { return bootCompleted; },
-      function fn(cb) {
-        i--;
-        if (i < 0) return cb(new Error('device didn\'t reboot within 90 seconds'));
-        if (i % 5 === 0) logger.debug('still waiting for reboot.');
-        this.shell("getprop sys.boot_completed", function (err, stdout) {
-          if (err) return cb(err);
-          bootCompleted = '1' === stdout.trim();
-          setTimeout(cb, 1000);
-        });
-      }.bind(this),
-      cb
-    );
-  }.bind(this));
-};
-
-ADB.getAdbServerPort = function () {
-  return process.env.ANDROID_ADB_SERVER_PORT || 5037;
-};
-
-module.exports = ADB;
+// transpile:mocha
+
+import _ from 'lodash';
+import path from 'path';
+import methods from './tools/index.js';
+
+class ADB {
+
+ constructor (opts = {}) {
+   if (typeof opts.sdkRoot === "undefined") {
+     opts.sdkRoot = process.env.ANDROID_HOME || '';
+   }
+   const defaults = {
+     sdkRoot: null,
+     udid: null,
+     appDeviceReadyTimeout: null,
+     useKeystore: null,
+     keystorePath: null,
+     keystorePassword: null,
+     keyAlias: null,
+     keyPassword: null,
+     adb: {path: "adb", defaultArgs: []},
+     tmpDir: null,
+     curDeviceId: null,
+     emulatorPort : null,
+     logcat: null,
+     binaries: {},
+     instrumentProc: null,
+     javaVersion: null,
+     suppressKillServer: null,
+     jars: {}
+   };
+   Object.assign(this, defaults, opts);
+   if (opts.remoteAdbHost) {
+     this.adb.defaultArgs.push("-H", opts.remoteAdbHost);
+   }
+   if (opts.remoteAdbPort) {
+     this.adb.defaultArgs.push("-P", opts.remoteAdbPort);
+   }
+   this.initJars();
+ }
+
+ initJars () {
+   let tempJars = ['move_manifest.jar', 'sign.jar', 'appium_apk_tools.jar',
+                   'unsign.jar', 'verify.jar'];
+   for (let jarName of tempJars) {
+     this.jars[jarName] = path.resolve(__dirname, '../jars', jarName);
+   }
+   if (!this.javaVersion || parseFloat(this.javaVersion) < 1.7) {
+     this.jars["appium_apk_tools.jar"] = path.resolve(__dirname,
+                                                      '../jars',
+                                                      "appium_apk_tools_1.6.jar");
+   }
+ }
+
+ async createADB (opts) {
+   let adb = new ADB(opts);
+   return adb.getAdbWithCorrectAdbPath();
+ }
+}
+
+for (let [mthds, fn] of _.pairs(methods)) {
```

Comment:
it's not really `mthds` it's more like `mthd`, which is ambiguous because it could be the function itself. what about `name`? or `fnName` since you're using `fn`?